packages/ckeditor5-table/src/commands/setheaderrowcommand.ts

```diff
@@ -43,10 +43,16 @@ export default class SetHeaderRowCommand extends Command {
 	public override refresh(): void {
 		const tableUtils: TableUtils = this.editor.plugins.get( 'TableUtils' );
 		const model = this.editor.model;
+
 		const selectedCells = tableUtils.getSelectionAffectedTableCells( model.document.selection );
 		const isInTable = selectedCells.length > 0;
+		let isContentTable = true;
+
+		if ( isInTable ) {
+			isContentTable = isTableTypeContent( selectedCells );
+		}
```

Comment:
We should check if schema allows `headingRows` attribute instead of coupling those features.